src/codeflare_sdk/utils/generate_yaml.py

```diff
@@ -191,6 +202,82 @@ def update_nodes(
+    metadata["name"] = f"ca-secret-{cluster_name}"
+    metadata["namespace"] = namespace
+    metadata["labels"]["odh-ray-cluster-service"] = f"{cluster_name}-head-svc"
+    data = ca_secret_item.get("generictemplate", {}).get("data")
+    data["ca.key"], data["ca.crt"] = generate_cert.generate_ca_cert(365)
+
+
+def enable_local_interactive(resources, cluster_name, namespace):
+    rayclient_route_item = resources["resources"].get("GenericItems")[2]
+    ca_secret_item = resources["resources"].get("GenericItems")[3]
+    item = resources["resources"].get("GenericItems")[0]
+    update_rayclient_route(rayclient_route_item, cluster_name, namespace)
+    update_ca_secret(ca_secret_item, cluster_name, namespace)
+    # update_ca_secret_volumes
+    item["generictemplate"]["spec"]["headGroupSpec"]["template"]["spec"]["volumes"][0][
+        "secret"
+    ]["secretName"] = f"ca-secret-{cluster_name}"
+    item["generictemplate"]["spec"]["workerGroupSpecs"][0]["template"]["spec"][
+        "volumes"
+    ][0]["secret"]["secretName"] = f"ca-secret-{cluster_name}"
+    # update_tls_env
+    item["generictemplate"]["spec"]["headGroupSpec"]["template"]["spec"]["containers"][
+        0
+    ]["env"][1]["value"] = "1"
+    item["generictemplate"]["spec"]["workerGroupSpecs"][0]["template"]["spec"][
+        "containers"
+    ][0]["env"][1]["value"] = "1"
+    # update_init_container
+    command = item["generictemplate"]["spec"]["headGroupSpec"]["template"]["spec"][
+        "initContainers"
+    ][0].get("command")[2]
+
+    command = command.replace("deployment-name", cluster_name)
+
+    server_name = (
+        oc.whoami("--show-server").split(":")[1].split("//")[1].replace("api", "apps")
+    )
+
+    command = command.replace("server-name", server_name)
+
+    item["generictemplate"]["spec"]["headGroupSpec"]["template"]["spec"][
+        "initContainers"
+    ][0].get("command")[2] = command
+
+
+def disable_raycluster_tls(resources):
+    del resources["GenericItems"][0]["generictemplate"]["spec"]["headGroupSpec"][
+        "template"
+    ]["spec"]["volumes"]
+    del resources["GenericItems"][0]["generictemplate"]["spec"]["headGroupSpec"][
+        "template"
+    ]["spec"]["containers"][0]["volumeMounts"]
+    del resources["GenericItems"][0]["generictemplate"]["spec"]["headGroupSpec"][
+        "template"
+    ]["spec"]["initContainers"]
+    del resources["GenericItems"][0]["generictemplate"]["spec"]["workerGroupSpecs"][0][
+        "template"
+    ]["spec"]["volumes"]
+    del resources["GenericItems"][0]["generictemplate"]["spec"]["workerGroupSpecs"][0][
+        "template"
+    ]["spec"]["containers"][0]["volumeMounts"]
+    del resources["GenericItems"][0]["generictemplate"]["spec"]["workerGroupSpecs"][0][
+        "template"
+    ]["spec"]["initContainers"][1]
+    del resources["GenericItems"][3]  # rayclient route
+    del resources["GenericItems"][2]  # ca-secret
+
+
+def delete_route():
+    print("Delete Rayclient Route")
```

Comment:
Just noticed, does this function do anything / is there a reason for its existence?

Reply:
no it doesn't have any function probably just some pseudo code I wrote before.